compiler/src/dotty/tools/dotc/fromtasty/TastyFileUtil.scala

```diff
@@ -0,0 +1,41 @@
+package dotty.tools.dotc.fromtasty
+
+import java.nio.file.{Files, Path}
+import java.io
+
+import dotty.tools.dotc.core.NameKinds
+import dotty.tools.dotc.core.Names.SimpleName
+import dotty.tools.dotc.core.StdNames.nme
+import dotty.tools.dotc.core.tasty.{TastyUnpickler, TreePickler}
+
+object TastyFileUtil {
+
+  /** Get the class path and the class name including packages
+   *
+   *  If
+   *  ```scala
+   *    package foo
+   *    class Foo
+   *  ```
+   *  then `getClassName("./out/foo/Foo.tasty") returns `("./out", "foo.Foo")`
+   */
+  def getClassName(path: Path): (String, String) = {
+    assert(path.toString.endsWith(".tasty"))
+    assert(Files.exists(path))
+    val bytes = Files.readAllBytes(path)
+    val unpickler: TastyUnpickler = new TastyUnpickler(bytes)
+    val className =
+      unpickler.nameAtRef.contents.iterator.dropWhile {
```

Comment:
This is still fairly complicated and seems to rely on the name table being in a certain order. Maybe Tasty should have a special section at the beginning with just the fully qualified name of the class ?

Reply:
I seemed to have pushed from the wrong machine. I will push the actual reimplementation tonight.

Reply:
Now I pushed the correct code